src/cmd/policy.go

```diff
@@ -21,6 +24,87 @@ type regoInput struct {
 	Files []string `json:"files"`
 }
 
+type gitlabResponse struct {
+	Name        string             `json:"name,omitempty"`
+	Description string             `json:"description,omitempty"`
+	Language    string             `json:"language,omitempty"`
+	Languages   map[string]float64 `json:"languages,omitempty"`
+}
+
+type githubResponse struct {
+	Name        string             `json:"name,omitempty"`
+	Description string             `json:"description,omitempty"`
+	Language    string             `json:"language,omitempty"`
+	Languages   map[string]float64 `json:"languages,omitempty"`
+}
+
+type commonRepoMetadata struct {
+	Name        string             `json:"name,omitempty"`
+	Description string             `json:"description,omitempty"`
+	Language    string             `json:"language,omitempty"`
+	Languages   map[string]float64 `json:"languages,omitempty"`
+}
+
+type astMarshallable interface {
+	commonRepoMetadata | opslevel.Level
+}
+
+func githubResponseToRepoMetadata(response githubResponse) commonRepoMetadata {
```

Comment:
Tried to use a generic function here but it looks like you can't access fields if you use structs as the generic type 😞 
https://stackoverflow.com/questions/71530811/how-to-access-shared-field-in-structs-in-generics-in-go-1-18-i-get-error-type

Reply:
Oh yeah, that is much better.  I'll make those changes now.  You would think that would be my first thought after how much golang code I've read through recently.  lol